pkg/synopsysctl/cmd_migrate.go

```diff
@@ -324,24 +348,127 @@ func migrateBlackDuck(namespace string) error {
+				uploadCacheRCName = util.GetResourceName(blackDuckName, util.BlackDuckName, "uploadcache")
+				uploadCacheKeyPVCName = fmt.Sprintf("%s-blackduck-uploadcache-key", blackDuckName)
+				uploadCacheDataPVCName = fmt.Sprintf("%s-blackduck-uploadcache-data", blackDuckName)
+			}
+			// scale down zookeeper
+			err = scaleDownRC(blackDuckNamespace, zookeeperRCName)
+			if err != nil {
+				return err
+			}
+			// scale down upload cache
+			err = scaleDownRC(blackDuckNamespace, uploadCacheRCName)
+			if err != nil {
+				return err
+			}
+			err = migrateUploadCachePVCJob(blackDuckNamespace, blackDuckName, uploadCacheKeyPVCName, uploadCacheDataPVCName)
 			if err != nil {
 				return err
 			}
+
+			pvcs := []string{"blackduck-rabbitmq"}
+			if value, ok := blackDuck.Annotations["synopsys.com/created.by"]; ok && "pre-2019.6.0" == value {
+				pvcs = append(pvcs, "zookeeper-data", "zookeeper-datalog")
+			} else {
+				pvcs = append(pvcs, util.GetResourceName(blackDuckName, util.BlackDuckName, "zookeeper-data"), util.GetResourceName(blackDuckName, util.BlackDuckName, "zookeeper-datalog"))
+			}
+			for _, pvc := range pvcs {
+				err = removePVC(blackDuckNamespace, pvc, rabbitmqRCName)
+				if err != nil {
+					return err
+				}
+			}
 		}
 		log.Infof("successfully migrated Black Duck '%s' in namespace '%s'", blackDuckName, blackDuckNamespace)
 	}
 	return nil
 }
 
-func removePVC(blackDuckNamespace string) error {
-	_, err := util.GetPVC(kubeClient, blackDuckNamespace, "blackduck-rabbitmq")
+// removePVC removes the PVC
+func removePVC(namespace string, name string, rabbitmqRCName string) error {
+	_, err := util.GetPVC(kubeClient, namespace, name)
 	if err == nil {
-		return util.DeletePVC(kubeClient, blackDuckNamespace, "blackduck-rabbitmq")
+		if "blackduck-rabbitmq" == name {
+			// scale down rabbitmq
+			err = scaleDownRC(namespace, rabbitmqRCName)
+			if err != nil {
+				return err
+			}
+		}
+		return util.DeletePVC(kubeClient, namespace, name)
 	}
 	return nil
 }
 
+// migrateUploadCachePVCJob create a Kube job to migrate the upload cache key data to upload cache data PVC
+func migrateUploadCachePVCJob(namespace string, name string, uploadCacheKeyVolumeName string, uploadCacheDataVolumeName string) error {
+	uploadCacheKeyVolume := components.NewPVCVolume(horizonapi.PVCVolumeConfig{PVCName: uploadCacheKeyVolumeName, VolumeName: "dir-uploadcache-key"})
+	uploadCacheDataVolume := components.NewPVCVolume(horizonapi.PVCVolumeConfig{PVCName: uploadCacheDataVolumeName, VolumeName: "dir-uploadcache-data"})
+	migrateJob := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "migrate-upload-cache",
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:    "migrate-key",
+							Image:   busyBoxImage,
+							Command: []string{"sh", "-c", "cp -pr /tmp/keys /opt/blackduck/hub/blackduck-upload-cache"},
```

Comment:
should we add -f option in case the migrate is re-run?  Job might fail on re-run otherwise.

Reply:
done